dio/lib/src/options.dart

```diff
@@ -352,10 +363,20 @@ class Options {
 
   /// HTTP request headers.
   ///
-  /// The keys of the header are case-insensitive,
-  /// e.g.: content-type and Content-Type will be treated as the same key.
+  /// The equality of the header keys is case-insensitive,
+  /// e.g.: `content-type` and `Content-Type` will be treated as the same key.
   Map<String, dynamic>? headers;
 
+  /// Whether the header keys should be case-insensitive.
+  /// e.g.: `Content-Type` will be sent as `content-type` eventually.
+  ///
+  /// Defaults to true.
+  ///
+  /// This option WILL NOT take effect on these circumstances:
+  /// - XHR ([HttpRequest]) does not support handling this explicitly.
+  /// - The HTTP/2 standard only supports lowercase header keys.
+  bool? caseInsensitiveHeaders;
```

Comment:
The negation in the name sounds weird, dies it make sense to call it `caseSensitiveHeaders` and default to false?

Reply:
Updated.